Generator/Sources/NeedleFramework/Parsing/Processors/Processor.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// The processing error types.
+enum ProcessingError: Error {
```

Comment:
Doing this means we're losing info about the exception. We can leave as is, and then later add in new cases to this enum if we need it. 

The only concern is that we will have to add generic error types that have meanings that can apply to everyone that implements the `Processor` protocol.